src/Microsoft.OData.Core/TypeNameOracle.cs

```diff
@@ -86,12 +89,17 @@ internal static IEdmStructuredType ResolveAndValidateTypeFromTypeName(IEdmModel
             }
 
             // TODO: Clean up handling of expected types/sets during writing
-            var typeFromResource = (IEdmStructuredType)ResolveAndValidateTypeName(model, typeName, EdmTypeKind.None, /* expectStructuredType */ true, writerValidator);
+            var typeFromResource = ResolveAndValidateTypeName(model, typeName, EdmTypeKind.None, /* expectStructuredType */ true, writerValidator);

```

Comment:
>var [](start = 12, length = 3)

nit: can we take this opportunity to remove a var? hard to groc type in this case.